akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -208,17 +210,38 @@ private[io] object SelectionHandler {
       execute {
         new Task {
           def tryRun(): Unit = {
+            if (settings.TraceLogging) log.debug(s"Enabling $ops on $key")
             val currentOps = key.interestOps
             val newOps = currentOps | ops
             if (newOps != currentOps) key.interestOps(newOps)
           }
         }
       }
 
-    private def cancelKey(key: SelectionKey): Unit =
+    private def cancelKeyAndClose(key: SelectionKey, andThen: () ⇒ Unit): Unit =
+      execute {
+        new Task {
+          def tryRun(): Unit = {
+            Try(key.cancel())
+            Try(key.channel().close())
```

Comment:
Is it ok to do the same on Java 8 and 11 with this fix?

Reply:
Yes, that was the goal. We do a bit more than before on JDK 8. I guess, it's a good change anyway because it might fix https://github.com/akka/akka/issues/25605 as well.